core/src/impl/KokkosExp_IterateTileGPU.hpp

```diff
@@ -66,97 +66,42 @@ struct EmulateCUDADim3 {
 };
 #endif
 
+template <class Tag, class Functor, class... Args>
+KOKKOS_IMPL_DEVICE_FUNCTION
```

Comment:
Since this is part of some very optimization-sensitive code, we should probably forceinline this and the other `_tag_invoke*` helpers, though I don't know if we have a macro for this.